users/service.go

```diff
@@ -249,11 +281,11 @@ func (svc usersService) GenerateResetToken(ctx context.Context, email, host stri
 }
 
 func (svc usersService) ResetPassword(ctx context.Context, resetToken, password string) error {
-	email, err := svc.identify(ctx, resetToken)
+	ir, err := svc.identify(ctx, resetToken)
 	if err != nil {
 		return errors.Wrap(ErrUnauthorizedAccess, err)
 	}
-	u, err := svc.users.RetrieveByEmail(ctx, email)
+	u, err := svc.users.RetrieveByEmail(ctx, ir.email)
 	if err != nil || u.Email == "" {
```

Comment:
shouldn't it be here and the operator not or? 
Anyway, the next line is error checking IMHO it would be better to first check for the error then check if user email is empty, more readable.